diff -urN dist.orig/sphinxcontrib/inlinesyntaxhighlight.py dist/sphinxcontrib/inlinesyntaxhighlight.py
--- dist.orig/sphinxcontrib/inlinesyntaxhighlight.py	2017-09-16 13:27:20.825601390 +0900
+++ dist/sphinxcontrib/inlinesyntaxhighlight.py	2017-09-16 13:31:05.055603526 +0900
@@ -40,7 +40,7 @@
 PRE_DIV_RE = re.compile(r'\s*</pre></div>\s*$')
 
 def html_visit_literal(self, node):
-    env = self.settings.env
+    env = self.builder.env
 
     shall_highlight = False
 
diff -urN dist.orig/sphinxcontrib/scaladomain.py dist/sphinxcontrib/scaladomain.py
--- dist.orig/sphinxcontrib/scaladomain.py	2017-12-09 11:23:34.073819199 +0900
+++ dist/sphinxcontrib/scaladomain.py	2017-12-09 11:24:15.413819072 +0900
@@ -13,14 +13,13 @@
 __release__ = '0.1a1'
 
 from docutils import nodes
-from docutils.parsers.rst import directives
+from docutils.parsers.rst import directives, Directive
 
 from sphinx import addnodes
 from sphinx.directives import ObjectDescription
 from sphinx.domains import Domain, ObjType, Index
 from sphinx.roles import XRefRole
 from sphinx.util.docfields import Field, GroupedField, TypedField
-from sphinx.util.compat import Directive
 from sphinx.util.nodes import make_refnode
 
 class ScalaPackage(Directive):
